File: Python Banking System/SavingsAccount.py
from BankAccount import BankAccount
class SavingsAccount(BankAccount):
    def __init__(self, accountNumber, accountHolderName, balance, minimumBalance, interestRate, withdrawLimit):
        assert balance >= minimumBalance, "Initial balance must be greater than minimum balance"
        super().__init__(accountNumber, accountHolderName, balance)
        self._minimumBalance = minimumBalance
        self._interestRate = interestRate
        self._withdrawLimit = withdrawLimit

    def withdraw(self, amount):
        if (amount < self._withdrawLimit):
            if (self._balance - amount < self._minimumBalance):
                print("Withdraw denied, exceeds minimum balance")
            else:
                return super().withdraw(amount)
        else:
            print("Withdraw denied, exceeds withdraw limit")

    def applyInterest(self):
        self._balance = self._balance*(1+self._interestRate)
        print("-Interest rate applied-")

    def displayAccountDetails(self):
        print("Displaying information...")
        print(f"Account Number: {self._accountNumber}")
        print(f"Account Name: {self._accountHolderName}")
        print(f"Account Balance: {self._balance}")
        print(f"Minimum Balance: {self._minimumBalance}")
        print(f"Interest Rate: {self._interestRate}")
        print(f"Withdraw Limit: {self._withdrawLimit}")
        print()

File: Java Banking System/demo/src/main/java/com/example/SavingsAccount.java
package com.example;

public class SavingsAccount extends BankAccount{

    private int minimumBalance;
    private float interestRate;
    private int withdrawLimit;
    public SavingsAccount(int accountNumber, String accountHolderName, int balance, int minimumBalance, float interestRate, int withdrawLimit) {
        super(accountNumber, accountHolderName, balance);
        this.minimumBalance = minimumBalance;
        this.interestRate = interestRate;
        this.withdrawLimit = withdrawLimit;
    }
    
    @Override
    public void withdraw(int amount) {
        if (amount < withdrawLimit) {
            if (this.getBalance() - amount < this.minimumBalance) {
                System.out.println("Withdraw denied, exceeds minimum balance");
            }
            else {
                super.withdraw(amount);
            }
        }
        else {
            System.out.println("Withdraw denied, exceeds withdraw limit");
        }
    }

    public void applyInterest() {
        this.setBalance((int)(this.getBalance()*(1+this.interestRate)));
    }

    @Override
    public void displayAccountDetails() {
        System.out.println("Displaying information...");
        System.out.println("Account Number: " + this.getAccountNumber());
        System.out.println("Account Name: " + this.getAccountHolderName());
        System.out.println("Account Balance: " + this.getBalance());
        System.out.println("Minimum Balance: " + this.minimumBalance);
        System.out.println("Interest Rate: " + this.interestRate);
        System.out.println("Withdraw Limit: " + this.withdrawLimit + "\n");
    }

    public int getMinimumBalance(){
        return this.minimumBalance;
    }

    public float getInterestRate(){
        return this.interestRate;
    }

    public int getWithdrawLimit(){
        return this.withdrawLimit;
    }
}
